source/javascripts/components/common/RepoYmlStorageActions.tsx

```diff
@@ -9,40 +9,43 @@ type RepoYmlStorageActionsProps = {
 	appConfig: AppConfig | string;
 };
 
-const identity = (result: string): any => result;
+const identityParser = (result: string): any => result;
+
+const fetchOptions = {
+	method: "POST",
+	headers: {
+		"Content-Type": "application/yaml"
+	}
+};
 
 const useFormattedYml = (appConfig: AppConfig): string => {
 	const [yml, setYml] = useState(typeof appConfig === "string" ? appConfig : "");
-	const formatAppConfigRef = useRef<() => void>();
-	const { failed, result, call } = useMonolithApiCallback<string>(
-		"/api/cli/format",
-		{
-			method: "POST",
-			body: JSON.stringify(appConfig)
-		},
-		identity
-	);
+	const formatAppConfigRef = useRef<(options?: RequestInit) => void>();
+	const { failed, result, call } = useMonolithApiCallback<string>("/api/cli/format", fetchOptions, identityParser);
 
 	// NOTE: call function isn't referentially stable
```

Comment:
Is this still referentially unstable now that fetchOptions are moved out of `useFormattedYml `?

Reply:
Yes, as `useMonolithApiCallback` is still recreating the init object to add the csrf cookie.